### DISCUSSION TITLE
Define the structure and characteristics on Config sources

### DISCUSSION THREAD

**Discussion by Emily-Jiang** [eclipse-ee4j/config#66] [General]
Config source is the place where configuration properties are defined. We need to determine the characteristics for config source"
1. Does Config Source has a hierarchy? I will say "yes" so that the same properties can be overriden by different config sources.
2. What format of config sources do we want to support out of box: yaml, json, properties, xml?
3. what out of box config source should we support? environment variables, system properties, jakarta-config.properties?
What do you think?

**Comment by dmlloyd** [comment]
From the last couple of calls (other than the one I missed this week), I think we all more or less agree:

1. Config sources should be layered (I don't want to use the word "hierarchy" because this might be confusing with respect to tree structure).
2. We do want to support properties and YAML; I don't think we've agreed on any other formats (yet).
3. We haven't _specifically_ discussed support of environment variables or system properties yet, nor have we talked about the standard file name for properties-based config (though I think `jakarta-config.properties` is just fine).
